src/util/cert/libcrypto/cert.c

```diff
@@ -176,31 +177,116 @@ errno_t sss_cert_pem_to_der(TALLOC_CTX *mem_ctx, const char *pem,
 #define IDENTIFIER_NISTP384 "nistp384"
 #define IDENTIFIER_NISTP521 "nistp521"
 
+static int sss_ec_get_key(BN_CTX *bn_ctx, EVP_PKEY *cert_pub_key,
+                          EC_GROUP **_ec_group, EC_POINT **_ec_public_key)
+{
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+    int ret;
+    char curve_name[4096];
+    unsigned char pubkey[4096];
+    size_t len;
+    EC_GROUP *gr;
+    EC_POINT *pk;
+
+    ret = EVP_PKEY_get_utf8_string_param(cert_pub_key,
+                                         OSSL_PKEY_PARAM_GROUP_NAME,
+                                         curve_name, sizeof(curve_name), &len);
```

Comment:
NULL can be provided instead of `len` here.

Reply:
Replaced.